pkg/zonemgr/zone_file.go

```diff
@@ -2,180 +2,144 @@ package zonemgr
 }
 
-/*
-$ORIGIN mycluster.secondary.mydomain.com
-$TTL 1750
-@       IN      SOA ns1.mycluster.secondary.mydomain.com admin@secdns.com (
-*/
 func (zoneFile *ZoneFile) generateHeaderPrefix() {
-	zoneFile.headerPref = fmt.Sprintf("\n$ORIGIN %s \n$TTL %s \n@ IN SOA %s %s (", zoneFile.domainName, ttl,
-		zoneFile.userNameserverName, zoneFile.userAdminEmail)
-
-	//fmt.Println("======= prefix:\n", zoneFile.headerPref)
-
+	zoneFile.headerPref = fmt.Sprintf("$ORIGIN %s. \n$TTL %s \n@ IN SOA %s. %s. (", zoneFile.domain, ttl,
+		zoneFile.nameServerName, zoneFile.adminEmail)
 }
 
-/*
-									7200       ; refresh (2 hours)
-	                                3600       ; retry (1 hour)
-	                                1209600    ; expire (2 weeks)
-	                                3600       ; minimum (1 hour)
-	                                )
-	        IN      NS  ns1.mycluster.secondary.mydomain.com
-	        IN      A   1.2.3.4
-*/
 func (zoneFile *ZoneFile) generateHeaderSuffix() {
 	zoneFile.headerSuf = fmt.Sprintf(" %s %s %s %s)\n", refresh, retry, expire, ttl)
 
-	if zoneFile.userNameserverName != "" {
-		zoneFile.headerSuf += fmt.Sprintf("IN NS %s\n", zoneFile.userNameserverName)
-	}
-
-	if zoneFile.userNameserverIP != "" {
-		zoneFile.headerSuf += fmt.Sprintf("IN A %s\n", zoneFile.userNameserverIP)
+	if zoneFile.nameServerIP != "" {
+		zoneFile.headerSuf += fmt.Sprintf("IN NS %s.\n", zoneFile.nameServerName)
+		zoneFile.headerSuf += fmt.Sprintf("IN A %s\n", zoneFile.nameServerIP)
 	}
 }
 
-func (zoneFile *ZoneFile) generateHeader() (header string) {
+func (zoneFile *ZoneFile) generateHeader() string {
 	return zoneFile.headerPref + strconv.Itoa(zoneFile.soaSerial) + zoneFile.headerSuf
 }
 
-//////////// maps impl
+func (zoneFile *ZoneFile) updateVMIRecords(namespacedName k8stypes.NamespacedName, interfaces []v1.VirtualMachineInstanceNetworkInterface) bool {
+	key := fmt.Sprintf("%s_%s", namespacedName.Name, namespacedName.Namespace)
+	isUpdated := false
 
-func (zoneFile *ZoneFile) updateVMIRecords(vmiName string, vmiNamespace string, interfaces []v1.VirtualMachineInstanceNetworkInterface) (isUpdated bool, err error) {
-	key := fmt.Sprintf("%s_%s", vmiName, vmiNamespace)
-	isUpdated = false
-	//delete vmi records from the list
 	if interfaces == nil {
-		delete(zoneFile.vmiRecordsMap, key)
-		isUpdated = true
-	} else {
-		ifacesMap := zoneFile.vmiRecordsMap[key]
-		//if not exist or changed - override
-		if ifacesMap == nil || isIfaceListChanged(ifacesMap, interfaces) {
-			zoneFile.vmiRecordsMap[key] = buildInterfacesMap(vmiName, vmiNamespace, interfaces)
+		if zoneFile.vmiRecordsMap[key] != nil {
+			delete(zoneFile.vmiRecordsMap, key)
 			isUpdated = true
 		}
+	} else {
+		newRecords := buildARecordsArr(namespacedName.Name, namespacedName.Namespace, interfaces)
+		isUpdated = !isEqual(newRecords, zoneFile.vmiRecordsMap[key])
+		if isUpdated {
+			zoneFile.vmiRecordsMap[key] = newRecords
```

Comment:
I would keep the records sorted. So the order they are added to the zone file will be deterministic.

Reply:
done